### PR TITLE
icu4c 71.1

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/icu4c.rb
Comment (Bo98) [inline Formula/icu4c.rb:5]:
```suggestion
  version "71.1"
```

**Comment by Bo98** [comment]
Looks like there's no ICU-specific concerns here, just some build fixes for Xcode 13.3. I've already fixed most in my local branch.

As per last time, I will not be revision bumping texlive in this pull request.

Just a few things to clean up:

* MPD requires upstreaming a patch first - I'll do that soon.
* Ideally we'd also sort out the Qt5 issue described in #98765 but this is not a blocker - just nice to avoid two revision bumps.
* Decide on a merge technique - do we try the CODEOWNER method I suggested previously?

**Comment by Bo98** [comment]
This is waiting for a solution on how we built Qt5 on macOS 12.3+. I think that's the only blocker left.

**Comment by yurikoles** [comment]
`qt@5` was updated to 5.15.5 in #103229

**Comment by carlocab** [comment]
`12-arm64`:

```
Error: 8 failed steps!
brew install --verbose --build-bottle couchdb
brew install --verbose --build-bottle mysql
brew test --verbose node@12
brew install --verbose --build-bottle percona-xtrabackup
brew bottle --verbose --json postgresql --only-json-tab
brew audit qt@5 --git --skip-style
brew install --verbose --build-bottle swift
brew install --verbose --build-bottle vte3
```

`11-arm64`:
```
Error: 6 failed steps!
brew install --verbose --build-bottle couchdb
brew install --verbose --build-bottle mysql
brew install --verbose --build-bottle percona-xtrabackup
brew bottle --verbose --json postgresql --only-json-tab
brew audit qt@5 --git --skip-style
brew install --verbose --build-bottle vte3
```

Also lots of linkage failures, etc, for dependents.

**Comment by Bo98** [comment]
Hmm seems this has regressed over time rather than improved.

`couchdb` looks like a pre-existing erlang version problem. We should try fix that in a separate PR (so we can test easier if it works), merge that and try again here when done.

`mysql` apparently just refuses to build now if Homebrew boost is installed which is bizarre and we'll need to sort that out by whatever means necessary.

`percona-xtrabackup` conflict with `percona-server` yet again... I'll look into how that's happened but we may need to test that scenario more often.

`postgresql` is a `brew bottle` bug. I think we have a workaround in some versioned formula but we should probably fix it properly.

`swift` - I'll look into this since I've been maintaining this formula.

`vte3` - need to cherry-pick #98909 instead of revision bumping. I forgot to fixup and merge that PR a few weeks ago - sorry!

**Comment by Bo98** [comment]
> Also lots of linkage failures, etc, for dependents. 

Looking at the commits, everything before the letter L has gone missing. I probably still have them locally so I can add them back.

**Comment by Bo98** [comment]
> Also lots of linkage failures, etc, for dependents.

Looks like some commits went missing in the rebase. I probably still have them locally so I can add them back.

**Comment by SMillerDev** [comment]
Let's not cherry pick any fixes into this PR, that only delays them. If we need a fix for something we have a better chance in a new PR.

**Comment by carlocab** [comment]
I suggest dropping ICU support from `boost@1.76` (probably in a separate PR) to avoid having to rev bump it with every `icu4c` update.

**Comment by cho-m** [comment]
Monterey ARM failures:
```
Error: 8 failed steps!
brew test --verbose libxml2
brew bottle --verbose --json postgresql --only-json-tab
brew install --verbose --build-bottle couchdb
brew install --verbose --build-bottle gspell
brew test --verbose node@12
brew install --verbose --build-bottle tepl
brew install --verbose --build-bottle tracker
brew install --verbose --build-bottle vte3
```

---

- [ ] `libxml2` -- I think https://github.com/Homebrew/brew/pull/13533 may have broke `libxml2` as brew doesn't always pick a specific Python when multiple Python are in build environment (e.g. In other PRs, I think I've seen macOS prioritize `python@3.10` while Linux prioritize `python@3.9`). 
- [ ] `couchdb` -- we've seen this failure happen sporadically. As mentioned in #105876, it failed in 25.0.1 attempt, but went away in 25.0.2 attempt, and now returned in 25.0.3 attempt. No idea at this point, but thinking some issue with `rebar` v2 and Erlang 25, so maybe a temporary workaround of `erlang@24` may work.
- [ ] `gspell` -- needs glib-mkenums from `glib-utils` to build. Interestingly, configure detects that it is missing (`checking for glib-mkenums... no`), but build doesn't fail until it tries to run `$(GLIB_MKENUMS) --template ...`
- [ ] `node@12` -- same issue seen in `node@14` where the particular old copy of `gyp-mac-tool` has a `#!/usr/bin/env python`. In `node@14`, I went with a test-only PATH workaround (#104172), though wasn't sure if an env-script would have been better.
- [ ] `tepl` -- needs glib-mkenums from `glib-utils` to build
- [ ] `tracker`-- needs glib-mkenums from `glib-utils` to build
- [ ] `vte`-- needs glib-mkenums from `glib-utils` to build

**Comment by SMillerDev** [comment]
I'll start working on those. I'd argue however that those do not need to delay this merge since they're unrelated to icu4c.

**Comment by carlocab** [comment]
> I'd argue however that those do not need to delay this merge since they're unrelated to icu4c.

Normally I'd agree but merging this will break the bottles for those formulae, so we have to fix at least the ones modified in this PR.

**Comment by carlocab** [comment]
I merged #106485 in the meantime, since this doesn't look ready to merge.

**Comment by cho-m** [comment]
- Fixes for `gspell`, `tepl`, `tracker`, `vte3` in #106768
- Workaround for `couchdb` in #106767
- `node@10` will be dropped on Monterey due to Python 2 requirement, but it has been over a year since deprecation date so not too important.
- [ ] This leaves `postgresql` and `node@12` to avoid dropping bottles on Monterey. Probably better to do quick fix before next run.
- [ ] Linux had some other linkage failures in previous run https://github.com/Homebrew/homebrew-core/runs/7495559761?check_suite_focus=true
   - `poac`
      ```
      Full linkage --test poac output
        Missing libraries:
          unexpected (libicudata.so.70)
      ```
   - `unar`
      ```
      Full linkage --test unar output
        Missing libraries:
          unexpected (libicuuc.so.70)
      ```

**Comment by cho-m** [comment]
`mysql` build checks for Homebrew installation of `boost` and doesn't seem to provide a way around https://github.com/mysql/mysql-server/blob/mysql-8.0.30/CMakeLists.txt#L1930-L1935
```cmake
IF(APPLE)
  GET_FILENAME_COMPONENT(HOMEBREW_BASE ${HOMEBREW_HOME} DIRECTORY)
  IF(EXISTS ${HOMEBREW_BASE}/include/boost)
    FOREACH(SYSTEM_LIB ICU LIBEVENT LZ4 PROTOBUF ZSTD FIDO)
      IF(WITH_${SYSTEM_LIB} STREQUAL "system")
        MESSAGE(FATAL_ERROR
```

May be easier to leave broken linkage and fix afterward in PR without `boost`.

Would be nicer to improve build scripts to avoid failing if `boost` was installed.

**Comment by carlocab** [comment]
Or would be useful to have a
```
conflicts_with "boost" => :build
```
maybe. But I guess it's not too common. We could unlink `boost` manually in the formula...

**Comment by cho-m** [comment]
For other issues outside of `mysql`:
- [x] `node@12` - #106879
- [ ] `postgresql` - this seems to be related to the `strings` `-` option. Part of line is missing from output
   https://github.com/Homebrew/brew/blob/master/Library/Homebrew/keg_relocate.rb#L330

    ```console
    ❯ strings bin/postgres | rg PKG_CONFIG | tr ' ' '\n' | rg PKG_CONFIG
    'PKG_CONFIG_PATH=/opt/homebrew/opt/icu4c/lib/pkgconfig:/opt/homebrew/opt/openssl@1.1/lib/pkgconfig:/opt/homebrew/opt/krb5/lib/pkgconfig:/opt/homebrew/opt/lz4/lib/pkgconfig:/opt/homebrew/opt/readline/lib/pkgconfig'
    'PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/12'
    
    ❯ strings - bin/postgres | rg PKG_CONFIG | tr ' ' '\n' | rg PKG_CONFIG
    'PKG_CONFIG_PATH=/opt/homebrew/opt/icu4c/lib/pkgconfig:/opt/homebrew/opt/openssl@1.1/lib/pkgconfig:/opt/homebrew/opt/krb5/lib/pkgconfig:/opt/homebrew/opt/lz4/lib/pkgconfig:/opt/homebrew/opt/readline/lib/pkgconfig'
    'PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/opt/homebrew/Library/
    ```

**Comment by carlocab** [comment]
`llvm` now has linkage with `icu4c` on Linux now that `libxml2` depends on `icu4c`, which is a pretty huge nightmare.

I'm probably going to disable `libxml2` in `llvm` at the next rebuild.

**Comment by iMichka** [comment]
Rebase necessary for:
`mysql`, `r`,  `freeciv`, `libphonenumber`

**Comment by bayandin** [comment]
Since this PR already required a rebase I'm going to merge https://github.com/Homebrew/homebrew-core/pull/107797 with conflicting `mapnik`

**Comment by carlocab** [comment]
ARM Monterey failures:
```
Error: 10 failed steps!
brew install --verbose --build-bottle spidermonkey
brew bottle --verbose --json postgresql --only-json-tab
brew install --build-from-source spidermonkey
brew install --only-dependencies --verbose --build-bottle couchdb
brew install --verbose --build-bottle couchdb
brew install --verbose --build-bottle mysql
brew install --verbose --build-bottle node@12
brew install --verbose --build-bottle percona-server
brew install --verbose --build-bottle percona-xtrabackup
brew install --verbose --build-bottle widelands
```
Similar on ARM Big Sur, minus `widelands`. `node@12` is expected on Monterey, but not sure about Big Sur.

**Comment by carlocab** [comment]
This should help with `mysql`, and `percona-{server,xtrabackup}`: https://github.com/Homebrew/homebrew-test-bot/pull/820

Not sure how to fix `chakra` and `spidermonkey`.

**Comment by cho-m** [comment]
> `spidermonkey` seems to be a Python issue, as it's trying to install stuff directly into `HOMEBREW_PREFIX`. I think @cho-m had a look at this, but not sure if there's a solution. This causes the dependent `couchdb` to fail.

I'll take another look. I think an older `distutils` via `ENV["SETUPTOOLS_USE_DISTUTILS"] = "stdlib"` should work around issue. May be due to the new code that we had to patch running `brew` command since it is bypassing distutils.cfg and injecting Homebrew's prefix.

---

> `postgresql` hardcodes `PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/opt/homebrew/Library/`. Not sure why this hasn't come up before, but we may be able to get away with unsetting `PKG_CONFIG_LIBDIR`.

My previous comment applies here for `strings` output getting truncated in https://github.com/Homebrew/brew/blob/master/Library/Homebrew/keg_relocate.rb#L330
```console
❯ strings bin/postgres | rg PKG_CONFIG | tr ' ' '\n' | rg PKG_CONFIG
'PKG_CONFIG_PATH=/opt/homebrew/opt/icu4c/lib/pkgconfig:/opt/homebrew/opt/openssl@1.1/lib/pkgconfig:/opt/homebrew/opt/krb5/lib/pkgconfig:/opt/homebrew/opt/lz4/lib/pkgconfig:/opt/homebrew/opt/readline/lib/pkgconfig'
'PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/12'

❯ strings - bin/postgres | rg PKG_CONFIG | tr ' ' '\n' | rg PKG_CONFIG
'PKG_CONFIG_PATH=/opt/homebrew/opt/icu4c/lib/pkgconfig:/opt/homebrew/opt/openssl@1.1/lib/pkgconfig:/opt/homebrew/opt/krb5/lib/pkgconfig:/opt/homebrew/opt/lz4/lib/pkgconfig:/opt/homebrew/opt/readline/lib/pkgconfig'
'PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/opt/homebrew/Library/
```

**Comment by carlocab** [comment]
> My previous comment applies here for `strings -` output getting truncated in https://github.com/Homebrew/brew/blob/master/Library/Homebrew/keg_relocate.rb#L330
> 
> ```
> ❯ strings bin/postgres | rg PKG_CONFIG | tr ' ' '\n' | rg PKG_CONFIG
> 'PKG_CONFIG_PATH=/opt/homebrew/opt/icu4c/lib/pkgconfig:/opt/homebrew/opt/openssl@1.1/lib/pkgconfig:/opt/homebrew/opt/krb5/lib/pkgconfig:/opt/homebrew/opt/lz4/lib/pkgconfig:/opt/homebrew/opt/readline/lib/pkgconfig'
> 'PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/12'
> 
> ❯ strings - bin/postgres | rg PKG_CONFIG | tr ' ' '\n' | rg PKG_CONFIG
> 'PKG_CONFIG_PATH=/opt/homebrew/opt/icu4c/lib/pkgconfig:/opt/homebrew/opt/openssl@1.1/lib/pkgconfig:/opt/homebrew/opt/krb5/lib/pkgconfig:/opt/homebrew/opt/lz4/lib/pkgconfig:/opt/homebrew/opt/readline/lib/pkgconfig'
> 'PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/opt/homebrew/Library/
> ```

Weird. Doesn't happen for me:
```
❯ strings - bin/postgres | rg PKG_CONFIG | tr ' ' '\n' | rg PKG_CONFIG
'PKG_CONFIG_PATH=/usr/local/opt/icu4c/lib/pkgconfig:/usr/local/opt/openssl@1.1/lib/pkgconfig:/usr/local/opt/krb5/lib/pkgconfig:/usr/local/opt/lz4/lib/pkgconfig:/usr/local/opt/readline/lib/pkgconfig'
'PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/12'
```

**Comment by cho-m** [comment]
> Weird. Doesn't happen for me:
> 
> ```
> ❯ strings - bin/postgres | rg PKG_CONFIG | tr ' ' '\n' | rg PKG_CONFIG
> 'PKG_CONFIG_PATH=/usr/local/opt/icu4c/lib/pkgconfig:/usr/local/opt/openssl@1.1/lib/pkgconfig:/usr/local/opt/krb5/lib/pkgconfig:/usr/local/opt/lz4/lib/pkgconfig:/usr/local/opt/readline/lib/pkgconfig'
> 'PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/12'
> ```

I think it happens with particular bottles, like the ones created in this PR. Didn't seem to happen when we built `postgresql` in isolation (e.g. #108093). May need to look further into binaries to see what is happening.

**Comment by cho-m** [comment]
Linux build failures:
- `qt` - somehow ran out of space. didn't happen in previous runs:
    ```
    2022-08-17T16:21:56.9258911Z Error: An exception occurred within a child process:
    2022-08-17T16:21:56.9259333Z   Errno::ENOSPC: No space left on device @ fptr_finalize_flush - /github/home/bottles/logs/qt/05.cmake
    ```
- `webkitgtk` - looking for `gobject-introspection`. Maybe from dropping runtime dependency on `harfbuzz`:
    ```
    --   No package 'gobject-introspection-1.0' found
    CMake Error at Source/cmake/OptionsGTK.cmake:328 (message):
      GObjectIntrospection is needed for ENABLE_INTROSPECTION.
    ```

**Comment by carlocab** [comment]
On ARM: 

`couchdb` failed the linkage test because it linked with `openssl@3` for some reason. Probably has a `$HOMEBREW_PREFIX/opt/openssl` hardcoded somewhere. We can probably fix it after if Intel macOS goes well.

`texlive` also failed the linkage test, but that's expected.

**Comment by carlocab** [comment]
Also only on ARM Big Sur:
```
  qt@5:
    * Files were found with references to the Homebrew shims directory.
      The offending files are:
        mkspecs/qmodule.pri
  Error: 1 problem in 1 formula detected
```
Not sure why that didn't happen on Monterey.

**Comment by carlocab** [comment]
`postgresql@10` failed to build on Linux. Probably shouldn't block this.

**Comment by danielnachun** [comment]
> `postgresql@10` failed to build on Linux. Probably shouldn't block this.
> 
> ```
>   deparse.o: file not recognized: File truncated
>   collect2: error: ld returned 1 exit status
>   ../../src/Makefile.shlib:309: recipe for target 'postgres_fdw.so' failed
>   make[1]: *** [postgres_fdw.so] Error 1
> ```

It looks like a race condition, and definitely unrelated, so it should not be a blocker.

**Comment by danielnachun** [comment]
Also on Linux:
```
Full linkage --test couchdb output
  Missing libraries:
  unexpected (libcrypto.so.3)
```
Maybe `couchdb` is preferring `openssl@3` to `openssl@1.1`?  This would be a simple fix after this is merged and probably shouldn't block this.

**Comment by carlocab** [comment]
> Also on Linux:
> 
> ```
> Full linkage --test couchdb output
>   Missing libraries:
>   unexpected (libcrypto.so.3)
> ```
> 
> Maybe `couchdb` is preferring `openssl@3` to `openssl@1.1`? This would be a simple fix after this is merged and probably shouldn't block this.

Yes, the same thing is happening on macOS: https://github.com/Homebrew/homebrew-core/pull/98809#issuecomment-1233744248

**Comment by carlocab** [comment]
`no long build conflict` label didn't work for #109404 😞

This will need a rebase. But we should probably finish CI off to see if there's anything we can fix before another run.

**Comment by carlocab** [comment]
Removed the `no long build conflict label` so I can merge #109295. This already needs a rebase.

**Comment by carlocab** [comment]
There are a handful more linkage failures on Linux. I didn't add them here, because CI takes long enough:
```
brew linkage --test poac
brew linkage --test unar
brew linkage --test planck
brew linkage --test pyside@2
brew linkage --test texlive
brew linkage --test dcmtk
brew linkage --test dovecot
brew linkage --test fourstore
brew linkage --test samba
```
However, not all of these are `icu4c`-related. In particular:
- `pyside@2` cannot find Qt5 libraries ([test output](https://github.com/Homebrew/homebrew-core/runs/8116257943?check_suite_focus=true#step:11:5264))
- `fourstore` cannot find `libavahi-*` ([test output](https://github.com/Homebrew/homebrew-core/runs/8116257943?check_suite_focus=true#step:11:30383))

The rest are complaints about a missing `libicu*`. These can be rev-bumped separately.

`texlive` is handled in #109365. CI has finished there, and that can be merged once this is.

**Comment by carlocab** [comment]
Dropped some `revision` bumps while rebasing.

**Comment by carlocab** [comment]
Builds are green on ARM, but dependents `clickhouse-odbc` and `texlive` failed. The latter is expected, the former is because of
```
  ==> Installing clickhouse-odbc dependency: libiodbc
  Error: Cannot install libiodbc because conflicting formulae are installed.
    unixodbc: because both install `odbcinst.h`
  
  Please `brew unlink unixodbc` before continuing.
```
`test-bot` is supposed to handle the unlinking, so something went wrong there.

**Comment by SMillerDev** [comment]
I think we can just rebuild that one after the merge.

**Comment by carlocab** [comment]
By "that one", do you mean `clickhouse-odbc`? Does it need to be rebuilt? I don't know if it has linkage with `icu4c`.

**Comment by carlocab** [comment]
Audits for `zk`, `znc`, and `zorba` failed on Linux with:
```
  Error: GitHub API Error: Bad credentials
  HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
    https://github.com/settings/tokens
```
That step probably just took too long.

**Comment by SMillerDev** [comment]
> By "that one", do you mean clickhouse-odbc? Does it need to be rebuilt? I don't know if it has linkage with icu4c.

Yeah, but obviously without linkage that wouldn't be needed.

**Comment by carlocab** [comment]
Only `audit` failures due to token errors for formulae modified in this PR, so at least nothing should break from formulae modified here.

Some dependents will need rebuilding, but it should be possible to get those done quickly.

**Review by carlocab — DISMISSED** [review]
I think this is as ready as it's going to get.

Here's a merge checklist, which can be done when CI finishes:

- [x] Merge this (use self-hosted runner for upload through GitHub UI)
- [x] Merge #109365 ASAP, but not before this is merged
- [x] Revert dd5c480f2ff440049667217165e2fc5fcde11bf4 - #109735
- [ ] Rev bump poac - #109739
- [ ] Rev bump unar - #109740
- [ ] Rev bump planck - #109741
- [ ] Rev bump dcmtk - #109742
- [ ] Rev bump dovecot - #109743
- [x] ~~Rev bump fourstore - #109744~~
- [ ] Rev bump samba - #109745
- [x] Merge #109672

**Comment by carlocab** [comment]
Intel Monterey is done, with no surprises -- basically the same results as Catalina. Only Intel Big Sur left, which I expect in about two hours or so.

**Comment by carlocab** [comment]
```
==> Installing `skopeo` for upload...
==> Downloading https://ghcr.io/v2/homebrew/core/icu4c/manifests/71.1
curl: (22) The requested URL returned error: 404 Not Found
Error: skopeo: Failed to download resource "icu4c_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/icu4c/manifests/71.1
```
`skopeo` depends on `icu4c` on Linux. Looks like this needs to be done the old-fashioned way.

**Review by carlocab — APPROVED** [review]

**Comment by Bo98** [comment]
Ah, and that's not an issue on GitHub hosted uploads because skopeo is already installed.

Something to think about how to handle in the future - it's probably not worked since the libxml2 change.

**Comment by carlocab** [comment]
May be worth building a statically-linked `skopeo` in our container. We'd have to keep it up-to-date though.

**Comment by carlocab** [comment]
Doing `pr-pull` locally. Will push the commits to #109365.

**Comment by carlocab** [comment]
🚢

**Comment by carlocab** [comment]
Merge checklist, for reference (GitHub keeps hiding it): https://github.com/Homebrew/homebrew-core/pull/98809#pullrequestreview-1096423836

**Pull request closed without merging**
